### DISCUSSION TITLE
CTFNote development and my fork

### DISCUSSION THREAD

**Discussion by JJ-8** [TFNS/CTFNote#236] [General]
Hey,

First of all thank you so much for making CTFNote open source in the first place. It is amazing software and I have so much fun developing with it!

As you know I maintain a CTFNote fork and have done so for a really long time (couple of years). I have contributed much of it back to upstream over the past years but I am seeing a significant slowdown in upstream development. Furthermore, since I have made database migrations, I am forced to perform PRs in a specific order because otherwise the migration software will refuse to migrate. This makes my contributions to upstream serialized.

The delays in PR merges and the migration system mean I cannot make PRs for new features anymore. I have added tons of big features to my fork such as full Discord integration and synchronization, tags and recently a full UI redesign (and much much more smaller features) and there are even more features I want to implement in the future. I currently don't see a way for 441 commits to get into upstream in a reasonable time frame.

Therefore I am considering to detach my fork from upstream so it will become a separate project which can live its own lifetime. To avoid naming confusions I am also considering to change the name slightly and maybe update the logo (but I am not a designer). I wanted to let you know such that this wouldn't come as a surprise to you and to hear your thoughts. I am thinking of making this change soon.

**Comment by XeR** [comment]
Hi JJ-8, long time no talk.

Thanks a lot for your contributions to CTFNote over the past few years.
They have been greatly appreciated.

Thank you for letting us know.
The code is free software. It is encouraged to add new features and fix bugs.
Nothing *forces* you to contribute back to the upstream project.
It is GPL-licensed so there are caveats, but they don't apply here.

Saying that the development slowed down is a nice euphemism. :-)
It is totally understandable that your fork is much more advanced.

The main developer, @B-i-t-K, was working on a new version earlier this year.
It is almost ready™.

The new version uses a REST API instead of GraphQL and a plugin system.
It was developed with teams like yours in mind who want to integrate anything, like Cyberchef or Discord.

BitK would like to talk with you about it.
He made a whole new UI and would like to show you a demo.
If you are interested, he would also like to talk about the future of the project with you.

We use IRC to communicate, but he made a [Discord guild](https://discord.gg/UCwmTyUq) instead if you prefer.

**Comment by JJ-8** [comment]
v3.0.0 just released with all my commits brought to upstream :)
